packages/angular_devkit/build_angular/src/builders/browser-esbuild/stylesheets.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import type { BuildFailure, BuildOptions, OutputFile } from 'esbuild';
+import { build } from 'esbuild';
+import * as path from 'path';
+
+/**
+ * Determines if an unknown value is an esbuild BuildFailure error object thrown by esbuild.
+ * @param value A potential esbuild BuildFailure error object.
+ * @returns `true` if the object is determined to be a BuildFailure object; otherwise, `false`.
+ */
+function isEsBuildFailure(value: unknown): value is BuildFailure {
```

Comment:
I've seen this method in a couple of other places, maybe extract/export it?

Reply:
extracted a base bundle function into a separate file with this utility function